File: client-android/app/src/main/java/app/serenada/android/call/SignalingTransport.kt
package app.serenada.android.call

internal interface SignalingTransport {
    val kind: SignalingClient.TransportKind

    fun connect(
        host: String,
        onOpen: () -> Unit,
        onMessage: (SignalingMessage) -> Unit,
        onClosed: (String) -> Unit
    )

    fun send(message: SignalingMessage)

    fun close()

    fun resetSession() {}
}

File: client-ios/SerenadaCore/Sources/Signaling/SignalingTransport.swift
import Foundation

public enum TransportKind: String, CaseIterable {
    case ws
    case sse

    public var wireName: String { rawValue }
}

public protocol SignalingTransport: AnyObject {
    var kind: TransportKind { get }

    func connect(
        host: String,
        onOpen: @escaping () -> Void,
        onMessage: @escaping (SignalingMessage) -> Void,
        onClosed: @escaping (String) -> Void
    )

    func send(_ message: SignalingMessage)
    func close()
    func resetSession()
}

extension SignalingTransport {
    public func resetSession() {}
}
